### ISSUE TITLE
Typing error with tracks.

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#2894]
The code
```
s = video.color(0xff0000)
s = source(source.tracks(s).video)
output.sdl(s)
```
is raising the error
```
At ./bug.liq, line 3, char 11:
output.sdl(s)

Error 5: this value has type
  source(canvas('A)) (inferred at ./bug.liq, line 4, char 4-34)
but it should be a subtype of
  source(...)
```
which does not really make sense to me.

**Comment by toots** [comment]
This is a script error with a bad type error report. The syntax should be: `source({video = ...})`

Just pushed a commit fixing it that will display:
```
Error 5: this value has type
  source(canvas('A)) (inferred at ./GH2894.liq, line 3, char 6-36)
but it should be a subtype of
  source(video=_, _)
```

**Comment by toots** [comment]
Fixed in 687b1f23bc5875317ecd0d6bb06f8fa504011ec2
